src/util/templates/ChainedTemplateEngine.ts

```diff
@@ -0,0 +1,39 @@
+import type { Template, TemplateEngine } from './TemplateEngine';
+import Dict = NodeJS.Dict;
+
+/**
+ * Calls the given array of {@link TemplateEngine}s in the order they appear,
+ * feeding the output of one into the input of the next.
+ *
+ * The first engine will be called with the provided contents and template parameters.
+ * All subsequent engines will be called with no template parameter.
+ * Contents will still be passed along and another entry will be added for the body of the previous output.
+ */
+export class ChainedTemplateEngine<T extends Dict<any> = Dict<any>> implements TemplateEngine<T> {
+  private readonly contentsEngine: TemplateEngine<T>;
+  private readonly bodyEngines: TemplateEngine[];
+  private readonly bodyParam: string;
+
+  /**
+   * @param engines - Engines will be executed in the same order as the array.
+   * @param bodyParam - The name of the key used to pass the body of one engine to the next.
+   */
+  public constructor(engines: TemplateEngine[], bodyParam = 'body') {
+    if (engines.length === 0) {
+      throw new Error('At least 1 engine needs to be provided.');
+    }
+    this.contentsEngine = engines[0];
+    this.bodyEngines = engines.slice(1);
+    this.bodyParam = bodyParam;
```

Comment:
```suggestion
    this.firstEngine = engines[0];
    this.chainedEngines = engines.slice(1);
    this.renderedName = renderedName;
```

All very nitpicky, just to make the naming more general.